src/ui/debug_tools.rs

```diff
@@ -163,3 +178,66 @@ impl<'world, 'state, 'a, 'b, 'c> DebugRenderBackend
         }
     }
 }
+
+/// A plugin that draws the line where the Y-Sorting happens
+pub struct YSortDebugPlugin;
+
+impl Plugin for YSortDebugPlugin {
+    fn build(&self, app: &mut App) {
+        app.insert_resource(YSortDebug {
+            enabled: false,
+            stroke: Stroke::new(1.0, Color32::LIGHT_GRAY),
+        })
+        .add_system(draw_ysort_lines);
+    }
+}
+
+pub struct YSortDebug {
+    enabled: bool,
+    stroke: egui::Stroke,
+}
+
+/// Renders the ysort debug line
+fn draw_ysort_lines(
+    ysort_debug: Res<YSortDebug>,
+    mut egui_context: ResMut<EguiContext>,
+    query: Query<(&YSort, &Transform)>,
+    camera_query: Query<(&Camera, &GlobalTransform)>,
+) {
+    if !ysort_debug.enabled {
+        return;
+    }
+
+    if let Ok((camera, camera_transform)) = camera_query.get_single() {
+        for (ysort, transform) in query.iter() {
+            egui::CentralPanel::default()
+                .frame(egui::Frame::none())
+                .show(egui_context.ctx_mut(), |ui| {
+                    let mut a = transform.translation.clone();
+                    a.x += 16.;
+                    a.z = 0.;
+
+                    let mut b = transform.translation.clone();
+                    b.x += 16.;
```

Comment:
```suggestion
                    b.x -= 16.;
```

Hey @edgarssilva this is why the line wasn't displaying. Point `a` and `b` were the same point, so a line between them was invisible.

I found out by rendering a circle at each point instead of a line between them.

Reply:
Oh god I'm blind, thanks for picking it up.

Reply:
No problem, I've been there before. :D